### PR TITLE
codeintel: Update all periodic routines to use new goroutine utility

### PR DESCRIPTION
**Pull request by efritz**
Update the components in precise code intel services that periodically perform an action to use the new utilities introduced in https://github.com/sourcegraph/sourcegraph/pull/13260.

### CODE REVIEW

File: enterprise/cmd/precise-code-intel-worker/internal/commit-updater/updater.go [hunk 1]
```diff
@@ -17,72 +16,36 @@ import (
...
-		clock:    clock,
-		ctx:      ctx,
-		cancel:   cancel,
-		finished: make(chan struct{}),
+// Handle checks for dirty repositories and invokes the underlying updater on each one.
+func (u *Updater) Handle(ctx context.Context) error {
+	repositoryIDs, err := u.store.DirtyRepositories(ctx)
+	if err != nil {
+		log15.Error("Failed to retrieve dirty repositories", "err", err)
+		return nil
```
Comment (Strum355) [inline enterprise/cmd/precise-code-intel-worker/internal/commit-updater/updater.go:41]: Curious why returning nil on error? Does the background goroutine terminate on error?

Comment (efritz) [reply enterprise/cmd/precise-code-intel-worker/internal/commit-updater/updater.go]: Updated

File: enterprise/cmd/precise-code-intel-indexer/internal/index_manager/manager.go [hunk 2]
```diff
@@ -68,19 +50,23 @@ type ManagerOptions struct {
...
 	store            dbworkerstore.Store
 	options          ManagerOptions
 	clock            glock.Clock
 	indexers         map[string]*indexerMeta
-	dequeueSemaphore chan struct{}   // tracks available dequeue slots
-	m                sync.Mutex      // protects indexers
-	ctx              context.Context // root context passed to the database
-	cancel           func()          // cancels the root context
-	finished         chan struct{}   // signals that Start has finished
+	dequeueSemaphore chan struct{} // tracks available dequeue slots
```
Comment (Strum355) [inline enterprise/cmd/precise-code-intel-indexer/internal/index_manager/manager.go:65]: `chan` works here :) wonder how [golang.org/x/sync/semaphore](https://pkg.go.dev/golang.org/x/sync/semaphore?tab=doc) compares in performance and in function

**Review by Strum355 — APPROVED** [review]

File: enterprise/cmd/precise-code-intel-indexer/internal/index_manager/manager.go (hunk 2 above)
Comment (efritz) [reply enterprise/cmd/precise-code-intel-indexer/internal/index_manager/manager.go:65]: Some semantics are different. Here we don't want to block when trying to acquire. The x/sync/semaphore package maintains an ordered list of things that are waiting on the semaphore to wake when another user drops the lock.

Comment (efritz) [reply enterprise/cmd/precise-code-intel-indexer/internal/index_manager/manager.go:65]: Could use both but I think using the semaphore package may muck up the intent a bit.

**Pull request merged**